core/templates/dev/head/pages/exploration_player/ExplorationEngineService.js

```diff
@@ -287,6 +288,12 @@ oppia.factory('ExplorationPlayerService', [
           });
         }
       },
+      setCurrentStateName: function(stateName) {
```

Comment:
All the state name stuff should be internal to ExplorationEngineService, you shouldn't need to expose an external API for setting it.

Reply:
What about when navigating through cards by arrows? Then, wouldn't stateName have to be set explicitly?

Reply:
I think ConversationSkinService or TranscriptService would maintain an index of the current card.

Reply:
(i.e. the index of the card in the list, not the state name. After all, that's what is really being navigated through with the arrows. When you have pre-tests, it's still possible to use arrow navigation despite them not having state names.)